src/scaffoldmaker/utils/tubemesh.py

```diff
@@ -156,7 +156,7 @@ def warpSegmentPoints(xList, d1List, d2List, segmentAxis,
                 vectorToFirstNode = [xRot1[c] - centroidRot[c] for c in range(3)]
                 if vector.magnitude(vectorToFirstNode) > 0.0:
                     cp = vector.crossproduct3(vector.normalise(vectorToFirstNode), sd2[nAlongSegment])
-                    if vector.magnitude(cp) > 0:
+                    if vector.magnitude(cp) > 1e-12:
```

Comment:
Don't hardcode magic numbers such as these in reusable utility library code. It will not work for all scales of models. Can you determine a tolerance based on ranges of coordinates passed in.